doc/queries.md

```diff
@@ -340,6 +341,209 @@ Result:
 }
 ```
 
+<a name="aggs-nested"></a>
+
+### 4. Nested Aggregation
+Guppy supports nested aggregations (sub-aggregations) for fields. Currently Guppy only supports two-level-sub-aggregations.
+There are two types of nested aggregations that is supported by Guppy: terms aggregation and missing aggregation, user can mix-and-match the using of both aggregations.
+
+#### 4.1. Terms Aggregation
+Terms aggregation returns histogram for a single `field` an array of fields. It means for each of the `key` of the single `field`, what is the distribution of each element from the array of fields. Results are wrapped by keywords `field` and also `key` and `count` for that `field`, example:
+
+```
+query ($nestedAggFields: JSON) {
+  _aggregation {
+  subject (nestedAggFields: $nestedAggFields) {
+    project {
+      histogram {
+        key
+        count
+        termsFields {
+          field
+          terms {
+            key
+            count
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+This query requires a `JSON` format variable `nestedAggFields`, which contains an array of `termsFields`. For example:
+
+```
+{
+  nestedAggFields: {
+    termsFields: [
+      gender,
+      someNonExistingField
+    ]
+  }
+}
+```
+
+Result:
+
```

Comment:
I'm having trouble understanding where you specify which field to aggregate on (in the query)... i think a comment like this would help:
```suggestion
Here `internal-test-1` and `internal-test-2` are example values of field `XXX`.

```